### PR TITLE
Run duplication

### PR DESCRIPTION
**Pull request by xplosunn**

### CODE REVIEW

**Comment by rtfpessoa** [comment]
@xplosunn Let's not merge any more PRs until we fix the integration tests.

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala [hunk 1]
```diff
@@ -113,6 +113,16 @@ class AnalyseExecutor(toolInput: Option[String],
     analyser.metrics(metricsTool, metricsFilesTarget.directory, Some(metricsFilesTarget.readableFiles))
   }
 
+  private def duplication(
+    duplicationTool: DuplicationTool,
+    filesTarget: FilesTarget,
+    localConfigurationFile: Either[String, CodacyConfigurationFile]): Try[Set[DuplicationClone]] = {
+    val metricsFilesTarget =
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: metrics?

File: core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala [hunk 2]
```diff
@@ -51,6 +52,25 @@ class CodacyPluginsAnalyser extends Analyser[Try] {
...
   }
 
+  override def duplication(duplicationTool: DuplicationTool,
+                           directory: File,
+                           files: Set[Path],
+                           timeout: Option[Duration] = Option.empty[Duration]): Try[Set[DuplicationClone]] = {
+
+    val srcFiles = FilesTarget(directory, files, Set.empty[Path])
+
+    val result = duplicationTool.run(directory, srcFiles, timeout)
```
Comment (rtfpessoa) [inline core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala]: If we need a `FilesTarget` here why would we pass only the files and then re-create the object?

File: core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala [hunk 3]
```diff
@@ -51,6 +52,25 @@ class CodacyPluginsAnalyser extends Analyser[Try] {
...
+
+    result match {
+      case Success(res) =>
+        logger.info(s"Completed metrics for ${duplicationTool.name} with ${res.size} results")
```
Comment (rtfpessoa) [inline core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala]: Metrics?

File: core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala [hunk 4]
```diff
@@ -51,6 +52,25 @@ class CodacyPluginsAnalyser extends Analyser[Try] {
...
+      case Failure(e) =>
+        logger.error(e)(s"Failed metrics for ${duplicationTool.name}")
```
Comment (bmbferreira) [inline core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala]: Metrics?

File: core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala (hunk 2 above)
Comment (xplosunn) [reply core/src/main/scala/com/codacy/analysis/core/analysis/CodacyPluginsAnalyser.scala]: For the Analyser[T] (and consequently CodacyPluginsAnalyser) to have consistent signatures. I can settle of FilesTarget for all of them or I can recreate here.

**Pull request merged**